app/models/report.rb

```diff
@@ -27,27 +27,17 @@ def editable?(target_user)
     user == target_user
   end
 
-  def save_report_and_mentioning_reports
+  def save_or_update_report_and_mentionings(report_params = nil)
```

Comment:
メソッド名が長いので
```suggestion
  def save_report_and_mentionings(report_params = nil)
```

だけでもいいかも？